flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -254,7 +249,7 @@ public void testPartitionPromotion() throws Exception {
 		testPartitionRelease(
 			(jobId, partitionId, taskExecutor, taskExecutorGateway, partitionTracker) -> {
 				final CompletableFuture<Collection<ResultPartitionID>> releasePartitionsFuture = new CompletableFuture<>();
-				runInTaskExecutorThreadAndWait(taskExecutor, () -> partitionTracker.setPromotePartitionsConsumer(releasePartitionsFuture::complete));
+				partitionTracker.setPromotePartitionsConsumer(releasePartitionsFuture::complete);
```

Comment:
I think this is not 100% correct as the internal `TestingTaskExecutorPartitionTracker#promotePartitionsConsumer` field is not volatile. Hence, the `TaskExecutor` might read a stale value. I would suggest to either setup these things before creating the `TaskExecutor` or if this is not possible, then make the fields volatile.

Reply:
hasn't failed so far 🙈

Reply:
On a more serious note I wanted to push this back a bit since it looked like a bigger change, and I didn't want to delay the removal of this method.
But I think I may have a quick solution that I'll try tomorrow.